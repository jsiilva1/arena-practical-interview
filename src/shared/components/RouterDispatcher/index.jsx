import React from 'react';
import { Switch, Route } from 'react-router-dom';

import LazyImport from '../../utils/lazyImport';

const ReposPage = LazyImport({
  loader: () => import('../../../containers/Repos' /* webpackChunkName: 'app-chunk-repos-page' */),
});

export default (() => {
  return (
    <>
      <Switch>
        <Route exact path="/" component={ReposPage} />
      </Switch>
    </>
  );
});