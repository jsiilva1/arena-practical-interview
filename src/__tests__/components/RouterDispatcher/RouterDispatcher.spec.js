import React from 'react';
import { shallow } from 'enzyme';

import RouterDispatcher from '../../../shared/components/RouterDispatcher';

describe('RouterDispatcher Component', () => {
  it('should render the switch router correctly', () => {
    const wrapper = shallow(<RouterDispatcher />);
    
    expect(wrapper).toMatchSnapshot();
  });
});
