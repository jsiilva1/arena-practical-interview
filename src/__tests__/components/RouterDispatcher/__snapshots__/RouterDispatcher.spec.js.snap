// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouterDispatcher Component should render the switch router correctly 1`] = `
<Fragment>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/pulls/:creator/:repo"
    />
  </Switch>
</Fragment>
`;
